packages/mobile-native-mcp-server/src/workflow/nodes/abstractSchemaNode.ts

```diff
@@ -6,52 +6,45 @@
  */
 
 import z from 'zod';
-import { interrupt } from '@langchain/langgraph';
 import { BaseNode } from './abstractBaseNode.js';
-import {
-  MCP_WORKFLOW_TOOL_OUTPUT_SCHEMA,
-  WORKFLOW_TOOL_BASE_INPUT_SCHEMA,
-  WorkflowToolMetadata,
-  MCPToolInvocationData,
-} from '../../common/metadata.js';
+import { MCPToolInvocationData } from '../../common/metadata.js';
 import { Logger, createComponentLogger } from '../../logging/logger.js';
+import { ToolExecutor, LangGraphToolExecutor } from './toolExecutor.js';
 
-export abstract class AbstractSchemaNode<
-  TInputSchema extends typeof WORKFLOW_TOOL_BASE_INPUT_SCHEMA,
-  TResultSchema extends z.ZodObject<z.ZodRawShape>,
-  TOutputSchema extends
-    typeof MCP_WORKFLOW_TOOL_OUTPUT_SCHEMA = typeof MCP_WORKFLOW_TOOL_OUTPUT_SCHEMA,
-> extends BaseNode {
-  protected abstract readonly workflowToolMetadata: WorkflowToolMetadata<
```

Comment:
`workflowToolMetadata` also goes away. Specific instances of `WorkflowToolMetadata` will be instantiated within the `execute` method of nodes, on demand.